src/analysis/calculus/fderiv_measurable.lean

```diff
@@ -0,0 +1,455 @@
+    have yone : ∥y∥ ≤ 1 :=
+      le_trans (y_lt.le) (pow_le_one _ (by norm_num) (by norm_num)),
+    -- define the scale `k`.
+    obtain ⟨k, hk, h'k⟩ : ∃ (k : ℕ), (1/2) ^ (k + 1) < ∥y∥ ∧ ∥y∥ ≤ (1/2) ^ k :=
+      exists_nat_pow_near_of_lt_one yzero yone (by norm_num : (0 : ℝ) < 1/2)
+      (by norm_num : (1 : ℝ)/2 < 1),
+    -- the scale is large enough (as `y` is small enough)
+    have k_gt : n e < k,
+    { have : ((1:ℝ)/2) ^ (k + 1) < (1/2) ^ (n e + 1) := lt_trans hk y_lt,
+      rw pow_lt_pow_iff_of_lt_one (by norm_num : (0 : ℝ) < 1/2) (by norm_num) at this,
+      linarith },
+    set m := k - 1 with hl,
+    have m_ge : n e ≤ m := nat.le_pred_of_lt k_gt,
+    have km : k = m + 1 := (nat.succ_pred_eq_of_pos (lt_of_le_of_lt (zero_le _) k_gt)).symm,
+    rw km at hk h'k,
+    -- `f` is well approximated by `L e (n e) k` at the relevant scale
+    -- (in fact, we use `m = k - 1` instead of `k` because of the precise definition of `A`).
+    have J1 : ∥f (x + y) - f x - L e (n e) m ((x + y) - x)∥ ≤ (1/2) ^ e * (1/2) ^ m,
+    { apply le_of_mem_A (hn e (n e) m (le_refl _) m_ge).2.2,
+      { simp only [mem_closed_ball, dist_self],
+        exact div_nonneg (le_of_lt P) (zero_le_two) },
+      { simp [dist_eq_norm],
+        convert h'k,
+        field_simp,
+        ring_exp } },
+    have J2 : ∥f (x + y) - f x - L e (n e) m y∥ ≤ 4 * (1/2) ^ e * ∥y∥ := calc
+      ∥f (x + y) - f x - L e (n e) m y∥ ≤ (1/2) ^ e * (1/2) ^ m :
+        by simpa only [add_sub_cancel'] using J1
+      ... = 4 * (1/2) ^ e * (1/2) ^ (m + 2) : by { field_simp, ring_exp }
+      ... ≤ 4 * (1/2) ^ e * ∥y∥ :
+        mul_le_mul_of_nonneg_left (le_of_lt hk) (mul_nonneg (by norm_num) (le_of_lt P)),
+    -- use the previous estimates to see that `f (x + y) - f x - f' y` is small.
+    calc ∥f (x + y) - f x - f' y∥
+        = ∥(f (x + y) - f x - L e (n e) m y) + (L e (n e) m - f') y∥ :
+      by { congr' 1, simp, abel }
+    ... ≤ ∥f (x + y) - f x - L e (n e) m y∥ + ∥(L e (n e) m - f') y∥ :
+      norm_add_le _ _
+    ... ≤ 4 * (1/2) ^ e * ∥y∥ + 12 * ∥c∥ * (1/2) ^ e * ∥y∥ :
+      add_le_add J2
+        (le_trans (le_op_norm _ _) (mul_le_mul_of_nonneg_right (Lf' _ _ m_ge) (norm_nonneg _)))
+    ... = (4 + 12 * ∥c∥) * ∥y∥ * (1/2) ^ e : by ring
+    ... ≤ (4 + 12 * ∥c∥) * ∥y∥ * (ε / (4 + 12 * ∥c∥)) :
+      mul_le_mul_of_nonneg_left he.le
+        (mul_nonneg (add_nonneg (by norm_num) (mul_nonneg (by norm_num) (norm_nonneg _)))
+          (norm_nonneg _))
+    ... = ε * ∥y∥ : by { field_simp [ne_of_gt pos], ring } },
+  rw ← this.fderiv at f'K,
+  exact ⟨this.differentiable_at, f'K⟩
+end
+
+theorem differentiable_set_eq_D (hK : is_complete K) :
+  {x | differentiable_at 𝕜 f x ∧ fderiv 𝕜 f x ∈ K} = D f K :=
+subset.antisymm (differentiable_set_subset_D _) (D_subset_differentiable_set hK)
+
+end fderiv_measurable_aux
+
+open fderiv_measurable_aux
+
+lemma fderiv_mem_iff {f : E → F} {s : set (E →L[𝕜] F)} {x : E} :
```

Comment:
I think that this should be moved to `analysis/calculus/fderiv.lean`.

Reply:
Done